source/slang/slang-options.cpp

```diff
@@ -425,11 +403,58 @@ struct OptionsParser
         int             argc,
         char const* const*  argv)
     {
+       
         // Copy some state out of the current request, in case we've been called
         // after some other initialization has been performed.
         flags = requestImpl->getFrontEndReq()->compileFlags;
 
-        DiagnosticSink* sink = requestImpl->getSink();
+        DiagnosticSink* requestSink = requestImpl->getSink();
+
+        SourceManager* parentSourceManager = requestSink->getSourceManager();
+
+        // We need a new source manager to track our command line 'source'
+
+        SourceManager sourceManager;
+        sourceManager.initialize(parentSourceManager, parentSourceManager->getFileSystemExt());
+
+        // Why create a new DiagnosticSink?
+        // We *don't* want the lexer that comes as default (it's for Slang source!)
+        // We may want to set flags that are different
+        // We will need to use a new sourceManager that will just last for this parse and will map locs to
+        // source lines.
+        //
+        // The *problem* is that we still need to communicate to the requestSink in some suitable way.
+        //
+        // 1) We could have some kind of scoping mechanism (and only one sink)
+        // 2) We could have a 'parent' diagnostic sink, that if we set we route output too
+        // 3) We use something like the ISlangWriter to always be the thing output too (this has problems because
+        // some code assumes the diagnostics are accessible as a string)
+        //
+        // The solution used here is to have DiagnosticsSink have a 'parent' that also gets diagnostics reported to.
```

Comment:
This comment seems to reinforce the idea that we need/want to tease apart some of the myriad things `DiagnosticSink` does into distinct types so that we can more easily swap in custom "adapters" for stuff like this.